src/lib/message/txHashEncoder.ts

```diff
@@ -0,0 +1,303 @@
+import { ethers } from 'ethers'
+import { BigNumber } from '@ethersproject/bignumber'
+import { zeroPad } from '@ethersproject/bytes'
+import { L1ToL2Message } from './L1ToL2Message'
+import { ArbSdkError } from '../dataEntities/errors'
+
+export class TxHashEncoder extends L1ToL2Message {
+  /**
+   * ArbitrumDepositTx has type 0x64.
+   * @param type //type of the tx
+   * @param l2ChainId
+   * @param fromAddress
+   * @param messageNumber
+   * @param destAddress
+   * @param l1Value
+   * @returns
+   */
+  public static calculateArbitrumDepositTxHash(
+    type: number,
```

Comment:
@yahgwai: About [this comment](https://github.com/OffchainLabs/arbitrum-sdk/pull/165#discussion_r964618035), the reason for passing the type is to be able to double check if we're using the right function for encoding. It should be already clear based on the function names but it adds a safety check. 

I can def remove it if you think it's redundant though

Reply:
nevermind this ^ it's totally restructured now